### ISSUE TITLE
Single components are always populated with their fields

### ISSUE THREAD

**Issue by Refrezsh** [strapi/strapi#16111]
### Required System information

- Node.js version: 18.5.0
- NPM version: 9.5.0
- Strapi version: 4.8.1
- Database: MySQL
- Operating system: Win 10
- Is your project Javascript or Typescript: Typescript

### Describe the bug

When you open a form for editing a single type or a collection type, the forms of built-in non-repeatable components are always expanded, even if they are not required.

### Steps to reproduce the behavior

1. Go to any type of content-manager edit form with Not repeatable component 
2. List to component form or reload page

### Expected behavior

The form of a non-repeatable component must not be open. At least for optional components. There was no such behavior before version 1.8.0. 
It creates an inconvenience when a component is optional, but it has one or more mandatory fields, and when I save the form, I get an error of unfilled fields in an optional component that I didn't even plan to add. And after that I need to disable it first and then save it.

### Screenshots

When opening or reloading the form is always open
![Снимок экрана 2023-03-16 в 17 47 15](https://user-images.githubusercontent.com/80524240/225676776-49036385-2fe6-4b28-a2b7-97f4361a1b6c.png)

And for the optional component before the upgrade it was like this
![Снимок экрана 2023-03-16 в 17 47 02](https://user-images.githubusercontent.com/80524240/225677023-32d56aa1-946a-4c1c-b0e6-425b21158095.png)

**Comment by vlowe85** [comment]
Same issue here, as mentioned on discord

> I just upgraded strapi today from 4.5.3 to 4.8.2

I have noticed an annoying change in behaviour

When creating or editing an article, it is trying to pre-populate a dynamic component like seen in the first picture. This is an optional component which has a required field if chosen to add. As it has been pre-populated pressing save fails due to missing required field within the component, now i have to click the delete button top right of the component to get rid of it.

Previously on adding or creating an article you would see the + button to add the component optionally. Is this supposed to be happening now in the later version? can i stop it?
<img width="887" alt="Screenshot 2023-03-17 at 16 17 42" src="https://user-images.githubusercontent.com/17430502/225971947-c137a32a-cd5b-45d6-af72-2d1505ddc3fa.png">
<img width="886" alt="Screenshot 2023-03-17 at 16 17 48" src="https://user-images.githubusercontent.com/17430502/225971954-799140e1-3c66-4c87-98ad-833e4dd88ceb.png">

**Comment by andrewpmoore** [comment]
Yeah, I'm seeing exactly the same, very frustrating when you try to edit anything on the page and it fails because the component has popped back in and needs deleting

**Comment by Refrezsh** [comment]
In general, this problem does not allow to work properly. 

I compared commits and changes and found a place that is potentially a bug. 

`packages/core/admin/admin/src/content-manager/components/EditViewDataManagerProvider/reducer.js`

Action: `INIT_FORM` `210: Line of code`

So problem with usage of fabric function `findAllAndReplace`. This function does some **magic** with the data and prepares it for the form state. A little above, there are 2 calls to this one, which would get the filter functions.

In general, the problem is that the function sets the values `250: Line of code` for components with null value.
```
{
                  ...data,
                  __temp_key__: 0,
}
```

If we add ignoreFalseyValues `236: Line of code`
```
   const mergedDataWithPreparedRelations = findAllRelationsAndReplaceWithEmptyArray(
          data,
          attributes,
          // Added { ignoreFalseyValues: true }
        );
```

And add negation for non-repeating components in fabric: `224: Line of code`

```
const findAllRelationsAndReplaceWithEmptyArray = findAllAndReplace(
          components,
          // Changed predicate
          (value) => value.type === 'relation' && !value.type === 'component',
          (_, { path }) => {
            if (state.modifiedData?.id === data.id && get(state.modifiedData, path)) {
              return get(state.modifiedData, path);
            }

            return [];
          }
        );
```

But this solution does not help with all pages. Some pages get undefined length error. Frankly, I didn't have time to go any deeper. I have finished writing a very temporary but working solution that just find empty components with possible nesting and modifies it as null. 

```
       // 258 Line of code
       const mergedDataWithTmpKeys = findComponentsAndReplaceWithTempKey(
          mergedDataWithPreparedRelations,
          attributes,
          { ignoreFalseyValues: true }
        );

        // Very temporary fix
        const clearTempKeysRecursively = (mergedData) => {
          Object.entries(mergedData).forEach(([key, objValue]) => {
            if(!objValue || typeof objValue !== "object" || Array.isArray(objValue)) {
              return;
            }

            if(objValue["__temp_key__"] !== 0) {
              return;
            }

            const valueKeys = Object.keys(objValue).length;
            if(valueKeys > 1) {
              clearTempKeysRecursively(objValue)
            }

            mergedData[key] = null;

            if(valueKeys > 1 && Object.values(objValue).every((v) => v === null)) {
              mergedData[key] = null;
            }
          });
        }

        // Call temp fix
        if(mergedDataWithTmpKeys) {
          clearTempKeysRecursively(mergedDataWithTmpKeys)
        }
```

**Comment by anthonyhexium** [comment]
Same issue here. Really annoying with pages who have publish mode activate.

Here a video example ...

https://user-images.githubusercontent.com/62936624/226198264-1e15f3d2-15e3-475f-ab6a-b65a729e85fa.mp4

**Comment by joshuaellis** [comment]
Thanks everyone, this was a regression and we're looking at solving this.

**Comment by anthonyhexium** [comment]
You're welcome @joshuaellis ! Hope this get solved soon !

**Comment by vlowe85** [comment]
@joshuaellis Just to add a note which may effect the severity. It is possible to save the content **with** the component populated where the components **required** fields have **not** been provided.

We are experienced client side errors due to bad saved data which should not be possible to save, this issue is negetivly impacting the integrity of our data. I am currently having to catch these and clean them up.

**Comment by joshuaellis** [comment]
I'm sorry i'm not sure what you're alluding to me doing, the issue has been fixed and will be released in the next version of strapi, if you need to have the fix now, you should be able to use the experimental build from our CRON job. If you're using that and experiencing a different issue, then I would encourage you to open a new issue for us to investigate into :)

**Comment by vlowe85** [comment]
Ok, i was just highlighting it has a detrmrntal effect on the data, not just an admin UI annoyance. I didnt know if that would prompt a higher severity and quicker hotfix.

**Comment by Xzandro** [comment]
> Ok, i was just highlighting it has a detrmrntal effect on the data, not just an admin UI annoyance. I didnt know if that would prompt a higher severity and quicker hotfix.

For us, this is not only an admin UI annoyance, we are straight up not able to edit content for now anymore, because we have A LOT of single components on a page and it would be just too annoying to constantly scan through the entire page and remove populated components. Also not sure when the error was introduced, but since there was a security issue in previous versions, using an earlier version might not be a good solution.
We also tried the nighly, but at least on Digital Ocean app platform, dependancy installation just did not work at all.

Do we are eagerly waiting for a quick release here.

**Comment by ddegooijer** [comment]
Same issue here, very annoying for customers.

**Comment by Martijn-Schoenmaker-Webbio** [comment]
Fixed in 4.9
